### PR TITLE
Implementing test for storage swap.

### PR DESCRIPTION
**Pull request by ricardomaraschini**
When user swaps storages on image registry configuration only one should
still be enabled on Status. Added test to swap an IPI provided storage
by an EmptyDir one and checking if only EmptyDir is still set on Status.

This is a test to cover #394 , it is a nice to have even if we decide not to 
backport the fix.

### CODE REVIEW

File: test/e2e/imageconfig_test.go [hunk 1]
```diff
@@ -94,3 +94,63 @@ func TestAdditionalTrustedCA(t *testing.T) {
...
+		metav1.GetOptions{},
+	); err != nil {
+		t.Fatal("unable to get image registry config")
+	}
+
+	// after swapping storage, only one should be set on status.
+	if config.Status.Storage.EmptyDir == nil {
+		t.Error("empty dir storage not set after update")
+	}
+	if config.Status.Storage.S3 != nil {
```
Comment (dmage) [inline test/e2e/imageconfig_test.go]: `if config.Status.Storage != imageregistryv1.ImageRegistryConfigStorage{EmptyDir: config.Status.Storage.EmptyDir} {`

Comment (ricardomaraschini) [reply test/e2e/imageconfig_test.go]: Done.

**Comment by adambkaplan** [comment]
/retitle Bug 1722878: Implementing test for storage swap

**Pull request merged**